apps/server/tests/tasks/planTask.e2e-spec.ts

```diff
@@ -244,92 +395,161 @@ describe('PlanTask (e2e)', () => {
+      name: 'A',
       category: preparationCategory,
       priority: 'medium',
       isFloat: true,
       user: user,
       chunkInfo: {
-        start: newDate(new Date(2022, 11, 14)),
+        start: newDate(new Date(2022, 12, 20)),
         minChunkDuration: moment.duration(1, 'hour'),
-        maxChunkDuration: moment.duration(3, 'hour'),
-        deadline: newDate(new Date(2022, 11, 20, 0, 0)),
-        estimation: moment.duration(4, 'hours'),
-        chillTime: moment.duration(15, 'minutes'),
+        maxChunkDuration: moment.duration(1.5, 'hour'),
+        deadline: newDate(new Date(2022, 12, 22, 0, 0)),
+        estimation: moment.duration(3, 'hours'),
+        chillTime: moment.duration(20, 'minutes'),
       },
     }).save();
 
-    // Those Chunks need to be thrown away by the algorithm and replanned
-    await Chunk.create({
-      duration: moment.duration(1, 'hour'),
-      task: prepareForLogicExam,
-      start: newDate(new Date(2022, 11, 16, 50)),
+    const B = await Task.create({
+      name: 'B',
+      category: preparationCategory,
+      priority: 'high',
+      isFloat: true,
+      user: user,
+      chunkInfo: {
+        start: newDate(new Date(2022, 12, 20)),
+        minChunkDuration: moment.duration(0.5, 'hour'),
+        maxChunkDuration: moment.duration(2, 'hour'),
+        deadline: newDate(new Date(2022, 12, 24, 0, 0)),
+        estimation: moment.duration(5, 'hours'),
+        chillTime: moment.duration(30, 'minutes'),
+      },
     }).save();
 
-    await Chunk.create({
-      duration: moment.duration(1, 'hour'),
-      task: prepareForLogicExam,
-      start: newDate(new Date(2022, 11, 19, 50)),
+    const C = await Task.create({
+      name: 'C',
+      category: preparationCategory,
+      priority: 'low',
+      isFloat: true,
+      user: user,
+      chunkInfo: {
+        start: newDate(new Date(2022, 12, 20)),
+        minChunkDuration: moment.duration(0.5, 'hour'),
+        maxChunkDuration: moment.duration(0.5, 'hour'),
+        estimation: moment.duration(2, 'hours'),
+        deadline: newDate(new Date(2022, 12, 25, 0, 0)),
+        chillTime: moment.duration(0, 'minutes'),
+      },
+    }).save();
+
+    await taskPlanner.planTask(A);
+    await taskPlanner.planTask(B);
+    await taskPlanner.planTask(C);
+  });
+
+  it('Start - Deadline (no chunks)', async () => {
+    const user = await User.create({
+      email: 'email@gmail.com',
+      password: 'password1234',
     }).save();
 
-    const PrepareForPhysicsExam = await Task.create({
-      name: 'Prepare for Physics',
+    // Insert two colors for categories
+    // Const tasks
+
+    // Add float tasks
+
+    const yellowColor = await Color.create({
+      hexCode: '#ffff00',
+    }).save();
+
+    const preparationCategory = await Category.create({
+      name: 'Preparation',
+      color: yellowColor,
+      user: user,
+    }).save();
+
+    // Now let's simulate already planned float task:
+
+    const A = await Task.create({
```

Comment:
I'm feeling like we could create a util function to not repeat those task creation processes.

Reply:
Tried to create some util, but it looks like overengineering because every test has its own dataset, sharing a really small amount of common data.

Reply:
I meant inside the single test suite. Just something like createTask(...args), where args are all needed for creating the task args omitting defaults.